crates/matrix-sdk/src/client/builder.rs

```diff
@@ -312,7 +312,11 @@ impl ClientBuilder {
                         None,
                         [MatrixVersion::V1_0].into_iter().collect(),
                     )
-                    .await?;
+                    .await
+                    .map_err(|e| match e {
+                        HttpError::ClientApi(_) => ClientBuildError::AutoDiscovery,
```

Comment:
This seems sensible, but don't we want to just move the inner error from `ClientApi` into `AutoDiscovery`? Why throw it away.

Reply:
It's most likely not going to contain anything interesting, but I wouldn't object to moving it 🤷🏼

Reply:
I agree, but it's more likely that someone will complain that the error has been thrown away than complain that the error has been preserved.

Reply:
You're right, the underlying error is preserved now.